lib/mocha.js

```diff
@@ -277,15 +283,67 @@ Mocha.prototype.invert = function() {
  * @param {Boolean} ignore
  * @return {Mocha}
  * @api public
- * @param {boolean} ignore
- * @return {Mocha}
  */
 Mocha.prototype.ignoreLeaks = function(ignore) {
   this.options.ignoreLeaks = Boolean(ignore);
   return this;
 };
 
 /**
+ * Enable or disable randomization of test execution order within suites.
+ *
+ * @param {(boolean|number|string)} [seed] Optional random seed.  Seed must be
+ *   a 32-bit unsigned integer, or a string which can convert to one.  If
+ *   `true`, a seed will be created.  If `false`, randomization will be
+ *   globally disabled (if it was enabled previously).
+ * @return {Mocha}
+ * @api public
+ */
+Mocha.prototype.randomize = function randomize(seed) {
+  if (seed !== false) {
+    if (!arguments.length || seed === true) {
+      // generate a number for seeding
+      seed = Mocha.seed();
+    } else if (!(typeof seed === 'number' && seed > 0
+      && seed <= MAX_SAFE_INTEGER && seed % 1 === 0)) {
+      // this would be a hexadecimal number represented as a string
+      if (/^(0[xX])?[a-fA-F0-9]{1,14}$'/.test(seed)) {
+        seed = parseInt(seed, 16);
+      } else {
+        // a decimal number as string, presumably
+        seed = parseInt(seed, 10);
+        if (isNaN(seed)) {
+          throw new Error('Invalid random seed. Expected unsigned 32-bit '
+            + 'integer; got: "' + seed + '"');
+        }
+      }
+    }
+
+    var engine = Random.engines.mt19937().seed(seed);
+    this.options.randomConfig = {
+      shuffleTests: function shuffleTests(tests) {
+        Random.shuffle(tests, engine);
+        return tests;
+      },
+      seed: seed,
+      hex: '0x' + Number(seed).toString(16) // for display
+    };
+  } else {
+    delete this.options.randomConfig;
+  }
+  return this;
+};
+
+/**
+ * Generate a random seed.
+ * @returns {number} Unsigned 32-bit integer
+ */
+Mocha.seed = function seed() {
+  return Random.integer(0, MAX_SAFE_INTEGER)(Random.engines.mt19937()
+    .autoSeed());
+};
```

Comment:
@boneskull, I don't think you understood @ScottFreeCode's question. The question was if `autoSeed()` generates a number that can be directly returned, i.e. if the following was possible

```js
Mocha.seed = function seed() {
  return Random.engines.mt19937().autoSeed();
};
```

The answer seems to be no, since `autoSeed` **doesn't return a seed;** rather it returns a **seeded random generator** using `Random.generateEntropyArray`. The generator is fed into the function returned by `Random.integer`. `generateEntropyArray` returns an array instead of a seed number.

There's also a minor inaccuracy in the JSDoc (`s/32/53/`), assuming we don't want to allow negative seeds (or random-js doesn't support them).

Altogether the code should IMO look like this:

```js
/**
 * Generate a random seed.
 * @returns {number} Unsigned 53-bit integer
 */
Mocha.seed = function seed() {
  /*
   * Create an auto-seeded random number generator.
   * We can't use `Random.generateEntropyArray` (used by `autoSeed`)
   * for this function since our seed is an integer.
   */
  var generator = Random.engines.mt19937().autoSeed();

  /*
   * Return a random integer that can be used as a seed, using the
   * random number generator we just created.
   */
  return Random.integer(0, MAX_SAFE_INTEGER)(generator);
};
```